fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DbFedoraToOcflObjectIndex.java

```diff
@@ -289,6 +315,7 @@ public void reset() {
         try {
             jdbcTemplate.update(TRUNCATE_MAPPINGS, Collections.emptyMap());
             jdbcTemplate.update(TRUNCATE_TRANSACTIONS, Collections.emptyMap());
+            this.mappingCache.invalidateAll();
```

Comment:
The `removeMapping()` method should immediately invalidate the key when it's a short-lived tx